prime-router/src/main/kotlin/fhirengine/translation/hl7/ValueSetMap.kt

```diff
@@ -0,0 +1,33 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import java.util.SortedMap
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.DEDUCTION,
+    include = JsonTypeInfo.As.PROPERTY,
+    defaultImpl = InlineValueSet::class
+)
+/**
+ * Interface for valueSet. Implement this interface to create a data provider
+ * that can return a SortedMap<String, String> at runtime.
+ */
+interface ValueSetMap<T> {
+    val configData: T
```

Comment:
`configData` seems to be being used for very different purposes in each of the concrete implementations of this interface, and I'd have trouble saying at the interface level what the property does. Maybe we can instead remove this and store whatever is needed for each class in the class itself. For instance, in `LookupTableValueSet` each of the three required fields can be their own properties, and the `LookupTableValueSetConfig` class can be removed. In `InlineValueSet`, instead of overriding this `configData`, the Json Property `values` can go directly into a property called `values`. Thoughts? Would be happy to talk through it further or pair on this tomorrow.

Reply:
I removed `configData` from the interface entirely in favor of each implementation storing the deserialized object directly. I'd prefer to keep `LookupTableValueSetConfig` as that is offloading input validation to Jackson.

Reply:
Ah yea that's fine, looks good, but with `configData` removed, you can also remove the generic type `T`!

Reply:
I further removed the generic type.